src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl

```diff
@@ -56,17 +60,31 @@ step `Δt`, gravitational acceleration `g`, and free surface at the `n`-th time-
 """
 function MGImplicitFreeSurfaceSolver(grid::AbstractGrid, 
                                      settings = nothing,
-                                     gravitational_acceleration = nothing, 
+                                     gravitational_acceleration = g_Earth,
+                                     reduced_dim = (false, false, false),
                                      placeholder_timestep = -1.0)
     arch = architecture(grid)
 
+    right_hand_side = Field{Center, Center, Nothing}(grid)
+
     # Initialize vertically integrated lateral face areas
     ∫ᶻ_Axᶠᶜᶜ = Field{Face, Center, Nothing}(with_halo((3, 3, 1), grid))
     ∫ᶻ_Ayᶜᶠᶜ = Field{Center, Face, Nothing}(with_halo((3, 3, 1), grid))
 
     vertically_integrated_lateral_areas = (xᶠᶜᶜ = ∫ᶻ_Axᶠᶜᶜ, yᶜᶠᶜ = ∫ᶻ_Ayᶜᶠᶜ)
 
     compute_vertically_integrated_lateral_areas!(vertically_integrated_lateral_areas)
+    coeffs = compute_matrix_coefficients(vertically_integrated_lateral_areas, grid, gravitational_acceleration)
+    matrix_constructors, diagonal, problem_size = matrix_from_coefficients(arch, right_hand_side, coeffs, reduced_dim)  
+
+    # Placeholder preconditioner and matrix are calculated using a "placeholder" timestep of -1.0
+    # They are then recalculated before the first time step of the simulation.
+
+    placeholder_constructors = deepcopy(matrix_constructors)
+    M = prod(problem_size)
+    update_diag!(placeholder_constructors, arch, M, M, diagonal, 1.0, 0)
+
+    matrix = arch_sparse_matrix(arch, placeholder_constructors)
     fill_halo_regions!(vertically_integrated_lateral_areas)
```

Comment:
should we fill the halos *before* we use `vertically_integrated_lateral_areas` in the matrix construction?

Reply:
Yes! Thank you :)